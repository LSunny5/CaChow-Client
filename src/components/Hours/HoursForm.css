/*Add hours page main body*/
.HFormContent {
    text-align: center;
    padding-bottom: 1rem;
    min-height: 91.5vh;
}

/*group holding all hour boxes*/
.hoursDisplay {
    width: 90%;
    margin: 0 auto;
}

/*box for each day*/
.hoursGroup {
    background: lightyellow;
    border-radius: 10px;
    padding: 0.5rem;
    margin: 0.5rem auto;
    border: 1px solid darkgreen;
}

/*day of the week for hours form*/
h3 {
    font-family: 'Cinzel', serif;
    font-size: 1.6rem;
    font-weight: bolder;
}

/*Time open and close label*/
.timeDesc {
    font-size: 1rem;
}

/*Box for time open and close each*/
.hoursSection {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

/********************************************************************/
/*For Ipads and larger screens */
@media only screen and (min-width: 600px) {
    /*group holding all hour boxes*/
    .hoursDisplay {
        width: 90%;
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
    }

    /*box for each day*/
    .hoursGroup {
        width: 45%;
        margin: 1rem auto;
    }
}

/********************************************************************/
/*For Laptops and larger screens */
@media only screen and (min-width: 1024px) {
    /*box for each day*/
    .hoursGroup {
        width: 30%;
        margin: 0.6rem auto;
    }
}