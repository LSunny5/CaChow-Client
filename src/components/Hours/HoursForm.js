import React from 'react';
import './HoursForm.css';
import config from '../../config';
import { NavLink } from 'react-router-dom';
import CachowContext from '../../CachowContext';
import TokenService from '../../services/token-service';
import PropTypes from 'prop-types';

class HoursForm extends React.Component {
    static contextType = CachowContext;

    handleUpdate = event => {
        const { title, value } = event.target;
        this.setState({ [title]: value.trim() });
    }

    handleSubmit = event => {
        event.preventDefault();
        let uname = localStorage.getItem('name');

        const newHours = {
            sun_open: event.target['SundayOpen'].value,
            sun_close: event.target['SundayClose'].value,
            mon_open: event.target['MondayOpen'].value,
            mon_close: event.target['MondayClose'].value,
            tues_open: event.target['TuesdayOpen'].value,
            tues_close: event.target['TuesdayClose'].value,
            wed_open: event.target['WednesdayOpen'].value,
            wed_close: event.target['WednesdayClose'].value,
            thu_open: event.target['ThursdayOpen'].value,
            thu_close: event.target['ThursdayClose'].value,
            fri_open: event.target['FridayOpen'].value,
            fri_close: event.target['FridayClose'].value,
            sat_open: event.target['SaturdayOpen'].value,
            sat_close: event.target['SaturdayClose'].value,
            hours_owner: uname,
        }
        
        fetch(`${config.APIENDPOINT}/hours`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`,
            },
            body: JSON.stringify(newHours),
        })
            .then(response => {
                if (!response.ok)
                    return response.json().then(e => Promise.reject(e))
                return response.json()
            })
            .then(h => {
                this.context.addHours(h);
                alert('Hours successfully added!');
                window.location.href = `${this.props.nextLocation}`;
            })
            .catch(error => {
                console.error({ error })
                alert('Error! ' + error);
            })
    }

    render() {


        const days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];

        return (
            <form className="addHourForm" onSubmit={this.handleSubmit}>
                <fieldset>
                    <label htmlFor="SundayOpen" className="inputLabel">Time Open on Sunday: </label>
                    <input
                        type="text"
                        id="SundayOpen"
                        name="SundayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="SundayClose" className="inputLabel">Time Close on Sunday: </label>
                    <input
                        type="text"
                        id="SundayClose"
                        name="SundayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="MondayOpen" className="inputLabel">Time Open on Monday: </label>
                    <input
                        type="text"
                        id="MondayOpen"
                        name="MondayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="MondayClose" className="inputLabel">Time Close on Monday: </label>
                    <input
                        type="text"
                        id="MondayClose"
                        name="MondayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="TuesdayOpen" className="inputLabel">Time Open on Tuesday: </label>
                    <input
                        type="text"
                        id="TuesdayOpen"
                        name="TuesdayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="TuesdayClose" className="inputLabel">Time Close on Tuesday: </label>
                    <input
                        type="text"
                        id="TuesdayClose"
                        name="TuesdayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="WednesdayOpen" className="inputLabel">Time Open on Wednesday: </label>
                    <input
                        type="text"
                        id="WednesdayOpen"
                        name="WednesdayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="WednesdayClose" className="inputLabel">Time Close on Wednesday: </label>
                    <input
                        type="text"
                        id="WednesdayClose"
                        name="WednesdayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="ThursdayOpen" className="inputLabel">Time Open on Thursday: </label>
                    <input
                        type="text"
                        id="ThursdayOpen"
                        name="ThursdayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="ThursdayClose" className="inputLabel">Time Close on Thursday: </label>
                    <input
                        type="text"
                        id="ThursdayClose"
                        name="ThursdayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="FridayOpen" className="inputLabel">Time Open on Friday: </label>
                    <input
                        type="text"
                        id="FridayOpen"
                        name="FridayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="FridayClose" className="inputLabel">Time Close on Friday: </label>
                    <input
                        type="text"
                        id="FridayClose"
                        name="FridayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="SaturdayOpen" className="inputLabel">Time Open on Saturday: </label>
                    <input
                        type="text"
                        id="SaturdayOpen"
                        name="SaturdayOpen"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <label htmlFor="SaturdayClose" className="inputLabel">Time Close on Saturday: </label>
                    <input
                        type="text"
                        id="SaturdayClose"
                        name="SaturdayClose"
                        className="textField"
                        placeholder="ex. 12:00AM"
                        onChange={this.handleUpdate}
                    />
                    <br />
                    <div className="buttonBox">
                        <button type='submit' className="button">
                            Next
                        </button>
                        <NavLink className="button" to={`/account`}>
                            Cancel
                        </NavLink>
                    </div>
                </fieldset>
            </form>
        );
    };
}

HoursForm.propTypes = {
    nextLocation: PropTypes.string, 
}

export default HoursForm;