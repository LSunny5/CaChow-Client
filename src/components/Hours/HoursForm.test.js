import React from 'react';
import ReactDOM from 'react-dom';
import HoursForm from './HoursForm';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`HoursForm component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        const props = {
            nextLocation: '/account'
        };
        ReactDOM.render(
            <BrowserRouter>
                <HoursForm {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the RegistrationPage route', () => {
        const wrapper = shallow(<HoursForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});