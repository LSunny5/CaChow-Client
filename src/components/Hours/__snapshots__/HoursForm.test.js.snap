// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoursForm component renders the RegistrationPage route 1`] = `
<form
  className="addHourForm"
  onSubmit={[Function]}
>
  <fieldset>
    <label
      className="inputLabel"
      htmlFor="SundayOpen"
    >
      Time Open on Sunday: 
    </label>
    <input
      className="textField"
      id="SundayOpen"
      name="SundayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="SundayClose"
    >
      Time Close on Sunday: 
    </label>
    <input
      className="textField"
      id="SundayClose"
      name="SundayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="MondayOpen"
    >
      Time Open on Monday: 
    </label>
    <input
      className="textField"
      id="MondayOpen"
      name="MondayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="MondayClose"
    >
      Time Close on Monday: 
    </label>
    <input
      className="textField"
      id="MondayClose"
      name="MondayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="TuesdayOpen"
    >
      Time Open on Tuesday: 
    </label>
    <input
      className="textField"
      id="TuesdayOpen"
      name="TuesdayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="TuesdayClose"
    >
      Time Close on Tuesday: 
    </label>
    <input
      className="textField"
      id="TuesdayClose"
      name="TuesdayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="WednesdayOpen"
    >
      Time Open on Wednesday: 
    </label>
    <input
      className="textField"
      id="WednesdayOpen"
      name="WednesdayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="WednesdayClose"
    >
      Time Close on Wednesday: 
    </label>
    <input
      className="textField"
      id="WednesdayClose"
      name="WednesdayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="ThursdayOpen"
    >
      Time Open on Thursday: 
    </label>
    <input
      className="textField"
      id="ThursdayOpen"
      name="ThursdayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="ThursdayClose"
    >
      Time Close on Thursday: 
    </label>
    <input
      className="textField"
      id="ThursdayClose"
      name="ThursdayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="FridayOpen"
    >
      Time Open on Friday: 
    </label>
    <input
      className="textField"
      id="FridayOpen"
      name="FridayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="FridayClose"
    >
      Time Close on Friday: 
    </label>
    <input
      className="textField"
      id="FridayClose"
      name="FridayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="SaturdayOpen"
    >
      Time Open on Saturday: 
    </label>
    <input
      className="textField"
      id="SaturdayOpen"
      name="SaturdayOpen"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <label
      className="inputLabel"
      htmlFor="SaturdayClose"
    >
      Time Close on Saturday: 
    </label>
    <input
      className="textField"
      id="SaturdayClose"
      name="SaturdayClose"
      onChange={[Function]}
      placeholder="ex. 12:00AM"
      type="text"
    />
    <br />
    <div
      className="buttonBox"
    >
      <button
        className="button"
        type="submit"
      >
        Next
      </button>
      <NavLink
        className="button"
        to="/account"
      >
        Cancel
      </NavLink>
    </div>
  </fieldset>
</form>
`;
